### PR TITLE
BUG Fix creation of duplicate ParentID field on subtables of SiteTree

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes https://github.com/silverstripe/silverstripe-blog/pull/314

E.g. if Page has the Lumberjack extension (when SiteTree itself already has Hierarchy) then both Page.ParentID and SiteTree.ParentID will be independently created.

This is rather serious because certain queries could use different tables.

Fortunately, it seems that creating and moving site tree pages writes to the base SiteTree table, so removing this column doesn't break existing sites.

### CODE REVIEW

**Comment by tractorcow** [comment]
@micmania1 can you please review this?

**Comment by micmania1** [comment]
Does this fix prevent ParentID being added to both tables or does it just work around that? I haven't got time to test at the moment.

**Comment by UndefinedOffset** [comment]
It does appear to for me, I'm not seeing the column get added when this change is applied.

**Comment by tractorcow** [comment]
It prevents it.

File: code/extensions/Lumberjack.php [hunk 1]
```diff
@@ -81,13 +88,24 @@ public function stageChildren($showAll = false) {
...
+		$children = $baseClass::get()
+			->filter('ParentID', (int)$this->owner->ID)
+			->exclude('ID', (int)$this->owner->ID)
+			->setDataQueryParam(array(
+				'Versioned.mode' => $onlyDeletedFromStage ? 'stage_unique' : 'stage',
+				'Versioned.stage' => 'Live'
+			));
+
+		if(!$showAll) {
+            $children = $children->filter('ShowInMenus', 1);
```
Comment (micmania1) [inline code/extensions/Lumberjack.php]: can you clear up spacing please?

Comment (tractorcow) [inline code/extensions/Lumberjack.php]: Ah my bad. :D

**Comment by tractorcow** [comment]
Fixed indentation. ;)

**Comment by tractorcow** [comment]
Pretty sure that https://github.com/silverstripe/silverstripe-blog/issues/316 issue is related to this as well.

**Comment by UndefinedOffset** [comment]
Ya that was one of the symptoms I was seeing. It's caused because the parent doesn't exist more or less. The bug here tricks the core into thinking it wasn't set.

**Comment by micmania1** [comment]
This looks okay, but it does highlight how fragile this currently is. 

http://api.silverstripe.org/3.1/source-class-Hierarchy.html#548  - liveChildren is called directly here and I think the only reason its working at the moment is because Lumberjack is applied to a higher class than Hierarchy.

Looking at the extension points available, we could swap out stageChildren() with an augmentStageChildren() hook in the future. Unfortunately there isn't an augmentLiveChildren() hook, so we'd have to add this to framework and wait for 3.3 release.

**Comment by tractorcow** [comment]
https://github.com/silverstripe/silverstripe-lumberjack/releases/tag/1.1.3

**Comment by stevie-mayhew** [comment]
Thanks!

**Pull request merged**